CHANGELOG.md

```diff
@@ -11,8 +11,22 @@ and this library adheres to Rust's notion of
   abstract types `NotePlaintextBytes`, `NoteCiphertextBytes`,
   `CompactNotePlaintextBytes`, and `CompactNoteCiphertextBytes` to the `Domain`
   trait.
-- Moved the specific constants into the `Domain` trait implementations.
-
+- Removed the separate `NotePlaintextBytes` type definition (as it is now an
+  associated type).
+- Added new `parse_note_plaintext_bytes`, `parse_note_ciphertext_bytes`, and
+  `parse_compact_note_plaintext_bytes` methods to the `Domain` trait.
+- Updated the `note_plaintext_bytes` method of the `Domain` trait to return the
+  `NotePlaintextBytes` associated type.
+- Updated the `encrypt_note_plaintext` method of `NoteEncryption` to return the
+  `NoteCiphertextBytes` associated type of the `Domain` instead of the explicit
+  array.
+- Updated the `enc_ciphertext` method of the `ShieldedOutput` trait to return an
+  `Option` of a reference instead of a copy.
+- Moved the specific constants into the `Domain` trait implementations, while
+  keeping the original constants for backward compatibility.
```

Comment:
I don't think we should keep the original constants. this might create confusion.

Reply:
Done (removed the constants and updated CHANGELOG accordingly).